Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/BoneJCommand.java

```diff
@@ -31,14 +32,19 @@
 
     protected void reportUsage() {
         if (reporter == null) {
-            final PrefService prefService = context().getService(PrefService.class);
-            final PluginService pluginService = context().getService(PluginService.class);
-            final CommandService commandService = context().getService(CommandService.class);
-            reporter = UsageReporter.getInstance(prefService, pluginService, commandService);
+            initReporter();
         }
         reporter.reportEvent(getClass().getName());
     }
 
+    private void initReporter() {
+        final PrefService prefService = context().getService(PrefService.class);
+        final PluginService pluginService = context().getService(PluginService.class);
+        final CommandService commandService = context().getService(CommandService.class);
+        final LogService logService = context().getService(LogService.class);
+        reporter = UsageReporter.getInstance(prefService, pluginService, commandService, logService);
```

Comment:
What happens to the logs collected by `logService`? Are they stored anywhere, and if so, saved logs need to be cleared if a user opts-out of data collection.

Reply:
They are printed to a window in ImageJ. The same one that shows exceptions etc. I don't think they're stored, but I don't know for sure. I'll find out.

As a side note `UsageReporter` takes in the interface `Logger`, so if we're not happy what `LogService` does we can change the actual type (or hack our own thing).

Reply:
Forum drew blank so I did some digging: seems like [`StderrLogService`](https://github.com/scijava/scijava-common/blob/master/src/main/java/org/scijava/log/StderrLogService.java) is the default implementation of `LogService`. It does not store logs.